lib/removeUnusedElements.js

```diff
@@ -0,0 +1,235 @@
+'use strict';
```

Comment:
Does the uncompressed data get spliced out of the original buffer in all cases?

It's possible that the uncompressed attributes and a texture may be stored in a buffer, and because that texture's bufferView references the buffer the buffer won't get removed in `removeUnusedElements`.

Reply:
Currently yes. I have another branch that I think should fix this issue here:
https://github.com/FrankGalligan/gltf-pipeline/commits/fix_buffer_packing

I can merge that branch into this branch. I also have some other branches as well that I want to merge. Like a fix for JOINT_ATTRIBUTES not using integer, and a branch that will add support to go from Draco glTF -> glTF.

I was going to make PRs from them after this was landed as I think it would be easier to review, but I can merge them into this this PR if you want.

Reply:
Maybe open a PR for `fix_buffer_packing` that targets this branch? The others can wait for later.

Reply:
https://github.com/AnalyticalGraphicsInc/gltf-pipeline/pull/363